common/rpc/interceptor/namespace_validator.go

```diff
@@ -62,10 +62,11 @@ var (
 	errTaskTokenNamespaceMismatch = serviceerror.NewInvalidArgument("Operation requested with a token from a different namespace.")
 
 	allowedNamespaceStates = map[string][]enumspb.NamespaceState{
-		"/temporal.api.workflowservice.v1.WorkflowService/StartWorkflowExecution":           {enumspb.NAMESPACE_STATE_REGISTERED},
-		"/temporal.api.workflowservice.v1.WorkflowService/SignalWithStartWorkflowExecution": {enumspb.NAMESPACE_STATE_REGISTERED},
-		"/temporal.api.operatorservice.v1.OperatorService/DeleteNamespace":                  {enumspb.NAMESPACE_STATE_REGISTERED, enumspb.NAMESPACE_STATE_DEPRECATED, enumspb.NAMESPACE_STATE_DELETED},
-		"/temporal.api.nexusservice.v1.NexusService/DispatchNexusTask":                      {enumspb.NAMESPACE_STATE_REGISTERED},
+		api.WorkflowServicePrefix + "StartWorkflowExecution":           {enumspb.NAMESPACE_STATE_REGISTERED},
+		api.WorkflowServicePrefix + "SignalWithStartWorkflowExecution": {enumspb.NAMESPACE_STATE_REGISTERED},
+		api.OperatorServicePrefix + "DeleteNamespace":                  {enumspb.NAMESPACE_STATE_REGISTERED, enumspb.NAMESPACE_STATE_DEPRECATED, enumspb.NAMESPACE_STATE_DELETED},
+		api.AdminServicePrefix + "DescribeMutableState":                {enumspb.NAMESPACE_STATE_REGISTERED, enumspb.NAMESPACE_STATE_DEPRECATED, enumspb.NAMESPACE_STATE_DELETED},
```

Comment:
Would it be helpful to add the "Why?" here as a code comment?

Reply:
I thought about it while trying to parse how to fix the issue, but ultimately I noticed that the line makes more sense to me one it was written. I'd be curious to hear your take, but for me it was a sort of "Yea, the adminService should allow visibility for a workflow no matter its state. I'm surprised we didn't in the first place" 

Additionally, I think I allowing the code to be self descriptive in this case may be preferable to a comment that may grow out of sync later.

Reply:
> for me it was a sort of "Yea, the adminService should allow visibility for a workflow no matter its state. I'm surprised we didn't in the first place"

I didn't think it was 100% obvious. For example, when I delete my AWS account, I won't have access anymore, but they are very likely deleting the resources asynchronously in the background.

But at the same time, it's not surprising behavior either. So I'm fine with how it is (and we always have `git blame`).